test/jdk/com/sun/jdi/SingleStepCompilationTest.java

```diff
@@ -0,0 +1,236 @@
+        if (args.length != 0) {
+            throw new RuntimeException("Unexpected arguments passed to test");
+        }
+
+        // These are all needed for WhiteBoxAPI
+        newArgs[0] = "-Xbootclasspath/a:.";
+        newArgs[1] = "-XX:+UnlockDiagnosticVMOptions";
+        newArgs[2] = "-XX:+WhiteBoxAPI";
+
+        // In order to make sure the compilation is complete before we exit busyWork(),
+        // we don't allow background compilations.
+        newArgs[3] = "-XX:-BackgroundCompilation";
+
+        // Disabled tiered compilations so a new compilation doesn't start in the middle
+        // of executing busyWork(). (Might not really be needed)
+        newArgs[4] = "-XX:-TieredCompilation";
+
+        new SingleStepCompilationTest(newArgs).startTests();
+    }
+
+    /********** event handlers **********/
+
+    EventRequestManager erm;
+    BreakpointRequest bkptRequest;
+    StepRequest stepRequest;
+
+    // When we get a bkpt we want to disable the request and enable single stepping.
+    public void breakpointReached(BreakpointEvent event) {
+        System.out.println("Got BreakpointEvent: " + event);
+        EventRequest req = event.request();
+        req.disable();
+        stepRequest = erm.createStepRequest(mainThread,
+                                            StepRequest.STEP_LINE,
+                                            StepRequest.STEP_OVER);
+        stepRequest.enable();
+    }
+
+    public void stepCompleted(StepEvent event) {
+        System.out.println("Got StepEvent: " + event);
+        EventRequest req = event.request();
+        req.disable();
+    }
+
+    public void eventSetComplete(EventSet set) {
+        set.resume();
+    }
+
+    public void vmDisconnected(VMDisconnectEvent event) {
+        System.out.println("Got VMDisconnectEvent");
+    }
+
+    /********** test core **********/
+
+    protected void runTests() throws Exception {
+        /*
+         * Get to the top of main() to determine targetClass and mainThread
+         */
+        BreakpointEvent bpe = startToMain("TestTarg");
+        targetClass = (ClassType)bpe.location().declaringType();
+        mainThread = bpe.thread();
+        erm = vm().eventRequestManager();
+
+        // The BKPT_LINE is the line we will STEP_OVER
+        Location loc1 = findLocation(targetClass, TestTarg.BKPT_LINE);
+        bkptRequest = erm.createBreakpointRequest(loc1);
+        bkptRequest.enable();
+
+        try {
+            addListener(this);
+        } catch (Exception ex){
+            ex.printStackTrace();
+            failure("failure: Could not add listener");
+            throw new RuntimeException("SingleStepCompilationTest: failed", ex);
+        }
+
+        vm().resume();
+        waitForVMDisconnect();
+
+        System.out.println("done with loop");
+        removeListener(this);
+
+        /*
+         * Deal with results of test. If anything has called failure("foo")
+         * then testFailed will be true
```

Comment:
Nit: Need dots at the end of lines: 200, 207 and 228.

Reply:
Fixed.